wallet/src/de/schildbach/wallet/ui/WalletFragment.kt

```diff
@@ -0,0 +1,342 @@
+import de.schildbach.wallet.livedata.Resource
+import de.schildbach.wallet.livedata.Status
+import de.schildbach.wallet.ui.CheckPinDialog.Companion.show
+import de.schildbach.wallet.ui.InputParser.StringInputParser
+import de.schildbach.wallet.ui.MainActivity.Companion.REQUEST_CODE_SCAN
+import de.schildbach.wallet.ui.PaymentsFragment.Companion.ACTIVE_TAB_PAY
+import de.schildbach.wallet.ui.PaymentsFragment.Companion.ACTIVE_TAB_RECEIVE
+import de.schildbach.wallet.ui.VerifySeedActivity.Companion.createIntent
+import de.schildbach.wallet.ui.dashpay.CreateIdentityService.Companion.createIntentForRetry
+import de.schildbach.wallet.ui.dashpay.DashPayViewModel
+import de.schildbach.wallet.ui.scan.ScanActivity
+import de.schildbach.wallet.ui.send.SendCoinsInternalActivity
+import de.schildbach.wallet.ui.send.SweepWalletActivity
+import de.schildbach.wallet_test.R
+import kotlinx.android.synthetic.main.home_content.*
+import kotlinx.android.synthetic.main.quick_actions_layout.*
+import kotlinx.android.synthetic.main.sync_status_pane.*
+import org.bitcoinj.core.Coin
+import org.bitcoinj.core.PrefixedChecksummedBytes
+import org.bitcoinj.core.Transaction
+import org.bitcoinj.core.VerificationException
+import org.bitcoinj.wallet.Wallet
+import org.bitcoinj.wallet.listeners.WalletCoinsReceivedEventListener
+import org.bitcoinj.wallet.listeners.WalletCoinsSentEventListener
+import org.dash.wallet.integration.uphold.ui.UpholdAccountActivity
+
+class WalletFragment : Fragment(R.layout.home_content) {
+
+    private var clipboardManager: ClipboardManager? = null
+    private var blockchainState: BlockchainState? = null
+    private var syncComplete = false
+    private var dashPayViewModel: DashPayViewModel? = null
+    private var isPlatformAvailable = false
+    private var noIdentityCreatedOrInProgress = true
+    private var retryCreationIfInProgress = true
+
+    private val walletApplication by lazy { WalletApplication.getInstance() }
+    private val wallet by lazy { walletApplication.wallet }
+    private val config by lazy { walletApplication.configuration }
+    private val coinsSendReceivedListener = OnCoinsSentReceivedListener()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initView()
+        initViewModel()
+        clipboardManager = activity?.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager?
+
+        val appBar: View = app_bar
+        val params = appBar.layoutParams as CoordinatorLayout.LayoutParams
+        if (params.behavior == null) {
+            params.behavior = AppBarLayout.Behavior()
+        }
+        val behaviour = params.behavior as AppBarLayout.Behavior?
+        behaviour!!.setDragCallback(object : DragCallback() {
+            override fun canDrag(appBarLayout: AppBarLayout): Boolean {
+                //TODO: WalletFragment and WalletTransactionsFragment can probably be merged.
```

Comment:
Future story?

Reply:
Yes, merging these two fragments would make the code cleaner and possibily smaller, also, these two fragments can't exist separately in our app.